build/makelib/golang.mk

```diff
@@ -170,9 +170,10 @@ go.validate: go.vet go.fmt
 go.vendor.lite: $(DEP)
 #	dep ensure blindly updates the whole vendor tree causing everything to be rebuilt. This workaround
 #	will only call dep ensure if the .lock file changes or if the vendor dir is non-existent.
-	@if [ ! -d $(GO_VENDOR_DIR) ] || [ ! $(DEP) ensure -no-vendor -dry-run &> /dev/null ]; then \
-		echo === updating vendor dependencies ;\
-		$(DEP) ensure ;\
+	@if [ ! -d $(GO_VENDOR_DIR) ]; then \
```

Comment:
i'm not sure i understand why the OR (`||`) expression did not work here, can you help me understand that?  i do see that we'll now call `make go.vendor` instead of `dep ensure` directly, but I don't understand why the compound conditional is now an if/else.

Reply:
The OR is fine however the `[ ! $(DEP) ensure -no-vendor -dry-run &> /dev/null ]` does not check the return code as we expect. we have to remove the `[]` and can no longer use the `||`. So I just reworked the logic. Calling `make go.vendor` avoid repeated blocks.

Reply:
got it, thank you for the explanation!